pytket_qir/qir.py

```diff
@@ -87,14 +93,28 @@
 }
 
 
+class WASMError(Exception):
+    pass
+
+
 class QirParser:
     """A parser class to return a pytket circuit from a QIR file."""
 
-    def __init__(self, file_path: str, gateset: Optional[CustomGateSet]) -> None:
+    def __init__(
+        self,
+        file_path: str,
+        gateset: Optional[CustomGateSet],
+        wasm_handler: Optional[WasmFileHandler],
+        wasm_int_type: Optional[types.Int] = types.Int(32),
+        qir_int_type: Optional[types.Int] = types.Int(64),
+    ) -> None:
         self.module = QirModule(file_path)
-        self.gateset: Union[GateSet, CustomGateSet] = (
-            gateset if gateset else PYQIR_GATES
-        )
+        self.gateset: CustomGateSet = gateset if gateset else PYQIR_GATES
+        self.wasm_handler = wasm_handler
+        assert wasm_int_type is not None
```

Comment:
Sure but would it make sense to have `wasm_int_type` as non-optional but defaulted parameter and `wasm_handler` still being optional ? But then that would mean rearranging the parameter list and potential API breaks. Let me see if this could be made with minimal disruptive changes.

Reply:
Done.